staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/stdlib.go

```diff
@@ -0,0 +1,616 @@
+//
+// Returns the positional index of the provided element in the list. Supports all equatable types.
+// If the element is not found, -1 is returned.
+//
+//     <list<T>>.indexOf(<T>) <int>
+//     <list<T>>.lastIndexOf(<T>) <int>
+//
+// Examples:
+//
+//     [1, 2, 2, 3].indexOf(2) // returns 1
+//     [1, 2, 2, 3].lastIndexOf(2) // returns 2
+//     [1].indexOf(2) // returns -1
+//
+//
+// find / findAll
+//
+// Returns substrings that match the provided regular expression. find returns the first match. findAll may optionally
+// be provided a limit. If the limit is set and >= 0, no more than the limit number of matches are returned.
+//
+//     <string>.find(<string>) <string>
+//     <string>.findAll(<string>) <list <string>>
+//     <string>.findAll(<string>, <int>) <list <string>>
+//
+// Examples:
+//
+//     "abc 123".find('[0-9]*') // returns '123'
+//     "abc 123".find('xyz') // returns ''
+//     "123 abc 456".findAll('[0-9]*') // returns ['123', '456']
+//     "123 abc 456".findAll('[0-9]*', 1) // returns ['123']
+//     "123 abc 456".findAll('xyz') // returns []
+//
+//
+// url
+//
+// Converts a string to a URL or results in an error if the string is not a valid URL.
+//
+//     url(<string>) <URL>
+//
+//
+// isURL
+//
+// Returns true if a string is a valid URL.
+//
+//     <string>.isURL() <bool>
+//
+//
+// getScheme / getUsername / getPassword / getHostname / getPort / getPath / getQuery / getFragment
+//
+// Return the parsed components of the URL.
+//
+//     <URL>.getScheme() <string>
+//     <URL>.getUsername() <string>
+//     <URL>.getPassword() <string>
+//     <URL>.getHostname() <string>
+//     <URL>.getPort() <string>
+//     <URL>.getPath() <string>
+//     <URL>.getQuery() <map <string>, <list <string>>
+//     <URL>.getFragment() <string>
+//
+// Examples:
+//     url('https://example.com/').getScheme() // returns 'https'
+//     url('https://user:pass@example.com/').getUsername() // returns 'user'
+//     url('https://user:pass@example.com/').getPassword() // returns 'pass'
+//     url('https://example.com:80/path').getHostname() // returns 'example.com'
+//     url('https://127.0.0.1:80/path').getHostname() // returns '127.0.0.1'
+//     url('https://[::1]:80/path').getHostname() // returns '::1'
+//     url('https://example.com:80/path').getPort() // returns '80'
+//     url('https://example.com/path').getPath() // returns '/path'
+//     url('https://example.com/path?k1=a&k2=b&k2=c').getQuery() // returns { 'k1': ['a'], 'k2': ['b', 'c']}
+//     url('https://example.com/path#overview').getFragment() // returns 'overview'
+//
+//
+func Stdlib() cel.EnvOption {
+	return cel.Lib(stdlib{})
+}
+
+type stdlib struct{}
+
+var paramA = decls.NewTypeParamType("A")
+var typeParamAList = []string{"A"}
+
+func (stdlib) CompileOptions() []cel.EnvOption {
+	return []cel.EnvOption{
+		cel.Declarations(
+			decls.NewFunction("isSorted",
+				decls.NewParameterizedInstanceOverload("list_a_is_sorted_bool",
```

Comment:
are these names user-facing or required to be stable?

Reply:
No, they're not user-facing. It's safe to change them.